ipc/pipe/pipe_factory.go

```diff
@@ -141,6 +143,7 @@ func (f *Factory) SpawnWorkerWithTimeout(ctx context.Context, cmd *exec.Cmd) (*w
 }
 
 func (f *Factory) SpawnWorker(cmd *exec.Cmd) (*worker.Process, error) {
+	//  worker.InitBaseWorker repeated 4 times, maybe we should divide worker creating and worker spawning?
```

Comment:
InitBaseWorker is an internal API. Needs to be in the internal folder in the future.
On the other hand, Spawn_* methods are used for the different communication approaches between the PHP worker and Go (like pipes, unix sockets, tcp).